### PR TITLE
Convert to functional component

### PR DESCRIPTION
**Pull request by cvanem**
**Resolves** https://github.com/techniq/mui-virtualized-table/issues/6

- [x] Separate logic and convert to functional component.
- [x] ~~Disable hover and mouse enter/exit events while dragging column resize handle.  This helps with drag performance.~~
- [x] Add prettier settings to package.json.
- [x] Update react-virtualized and react-draggable packages.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+import MultiGrid from 'react-virtualized/dist/commonjs/MultiGrid'
+import classNames from 'classnames'
+import {
+  Table,
+  TableCell,
+  TableFooter,
+  TablePagination,
+  TableSortLabel,
+  withStyles
+} from '@material-ui/core'
```
Comment (techniq) [inline src/index.js]: I typically import these individually to reduce bundle size, but webpack are typically pretty good nowadays.  See: https://material-ui.com/guides/minimizing-bundle-size/#how-to-reduce-the-bundle-size

File: package.json [hunk 2]
```diff
@@ -44,5 +44,12 @@
     "deploy-docs": "gh-pages -d docs",
     "preversion": "yarn build",
     "postpublish": "yarn build-docs && yarn deploy-docs"
+  },
+  "prettier": {        
+    "tabWidth": 2,
+    "semi": false,
```
Comment (techniq) [inline package.json]: I typically leave semi on in my projects (`singleQuote` is the only default I typically change).  I'm fine with this as long as it's consistent.

**Comment by cvanem** [comment]
@techniq Release 3.0.0-3, I don't think I have access to push to npm.  Can you take care of that?

**Comment by techniq** [comment]
Sure. I’ll try to remember to when I get home. I publish using “np” cli and will grant you npm perms to the project as well.

**Comment by techniq** [comment]
@cvanem Released using `np --no-cleanup` as `3.0.0-4` / `@next`.  I've also granted you write permissions to the package on npm so you can cut releases as well.  Thanks for the PR and maintenance of the project.

**Pull request merged**